edb/pgsql/delta.py

```diff
@@ -2074,50 +2074,74 @@ def get_columns(cls, pointer, schema, default=None):
 
     def rename_pointer(self, pointer, schema, context, old_name, new_name):
         if context:
+            # before proceeding with renaming, make sure that this is
+            # not a computable
+            pointer = self.get_object(schema, context)
+            source = pointer.get_source(schema)
+            if source is not None:
+                is_computable = source.get_is_derived(schema)
+
+                # potentially this is a link property, so source may
+                # be a link
+                if isinstance(source, s_links.Link):
+                    source = source.get_source(schema)
+
+                if source is not None:
+                    is_computable = (
+                        is_computable
+                        or source.is_view(schema)
+                        or source.get_union_of(schema)
+                    )
+            else:
+                is_computable = None
+
             old_name = sn.shortname_from_fullname(old_name)
             new_name = sn.shortname_from_fullname(new_name)
 
-            host = self.get_host(schema, context)
+            if not is_computable:
+                host = self.get_host(schema, context)
 
-            if host and old_name != new_name:
-                if (new_name.endswith('std::source') and
-                        not host.scls.generic(schema)):
-                    pass
-                else:
-                    old_col_name = common.edgedb_name_to_pg_name(old_name.name)
-                    new_col_name = common.edgedb_name_to_pg_name(new_name.name)
+                if host and old_name != new_name:
+                    if (new_name.endswith('std::source') and
```

Comment:
Please see if `==` works.  `endswith` isn't a reliable check.

Reply:
Apparently it works, at least all the tests pass.